pkg/kubernetes/reconcile_test.go

```diff
@@ -54,3 +57,111 @@ func TestExtract(t *testing.T) {
 		})
 	}
 }
+
+func mkobj(kind string, name string, ns string) map[string]interface{} {
+	if ns == "" {
+		return map[string]interface{}{
+			"kind":       kind,
+			"apiVersion": "apiversion",
+			"metadata": map[string]interface{}{
+				"name": name,
+			},
+		}
+	}
+
+	return map[string]interface{}{
+		"kind":       kind,
+		"apiVersion": "apiversion",
+		"metadata": map[string]interface{}{
+			"name":      name,
+			"namespace": ns,
+		},
+	}
+}
+
+func TestReconcileSorting(t *testing.T) {
```

Comment:
I think this would benefit from a test that ensures that kind takes priority over namespace (ie multi-kind, multi-namespace)

Reply:
Good point, will implement tomorrow.